tests/load/pipeline/test_scd2.py

```diff
@@ -0,0 +1,399 @@
+    )
+    def r(data):
+        yield data
+
+    # get validity column names
+    from_, to = (
+        DEFAULT_VALIDITY_COLUMN_NAMES
+        if validity_column_names is None
+        else map(SnakeCaseNamingConvention().normalize_identifier, validity_column_names)
+    )
+
+    # load 1 — initial load
+    dim_snap = [
+        {"nk": 1, "c1": "foo", "c2": "foo" if simple else {"nc1": "foo"}},
+        {"nk": 2, "c1": "bar", "c2": "bar" if simple else {"nc1": "bar"}},
+    ]
+    info = p.run(r(dim_snap))
+    ts_1 = get_load_package_created_at(p, info)
+    assert_load_info(info)
+    cname = "c2" if simple else "c2__nc1"
+    assert get_table(p, "dim_test", cname) == [
+        {from_: ts_1, to: ACTIVE_TS, "nk": 2, "c1": "bar", cname: "bar"},
+        {from_: ts_1, to: ACTIVE_TS, "nk": 1, "c1": "foo", cname: "foo"},
+    ]
+
+    # load 2 — update a record
+    dim_snap = [
+        {"nk": 1, "c1": "foo", "c2": "foo_updated" if simple else {"nc1": "foo_updated"}},
+        {"nk": 2, "c1": "bar", "c2": "bar" if simple else {"nc1": "bar"}},
+    ]
+    info = p.run(r(dim_snap))
+    ts_2 = get_load_package_created_at(p, info)
+    assert_load_info(info)
+    assert get_table(p, "dim_test", cname) == [
+        {from_: ts_1, to: ACTIVE_TS, "nk": 2, "c1": "bar", cname: "bar"},
+        {from_: ts_1, to: ts_2, "nk": 1, "c1": "foo", cname: "foo"},
+        {from_: ts_2, to: ACTIVE_TS, "nk": 1, "c1": "foo", cname: "foo_updated"},
+    ]
+
+    # load 3 — delete a record
+    dim_snap = [
+        {"nk": 1, "c1": "foo", "c2": "foo_updated" if simple else {"nc1": "foo_updated"}},
+    ]
+    info = p.run(r(dim_snap))
+    ts_3 = get_load_package_created_at(p, info)
+    assert_load_info(info)
+    assert get_table(p, "dim_test", cname) == [
+        {from_: ts_1, to: ts_3, "nk": 2, "c1": "bar", cname: "bar"},
+        {from_: ts_1, to: ts_2, "nk": 1, "c1": "foo", cname: "foo"},
+        {from_: ts_2, to: ACTIVE_TS, "nk": 1, "c1": "foo", cname: "foo_updated"},
+    ]
+
+    # load 4 — insert a record
+    dim_snap = [
+        {"nk": 1, "c1": "foo", "c2": "foo_updated" if simple else {"nc1": "foo_updated"}},
+        {"nk": 3, "c1": "baz", "c2": "baz" if simple else {"nc1": "baz"}},
+    ]
+    info = p.run(r(dim_snap))
+    ts_4 = get_load_package_created_at(p, info)
+    assert_load_info(info)
+    assert get_table(p, "dim_test", cname) == [
+        {from_: ts_1, to: ts_3, "nk": 2, "c1": "bar", cname: "bar"},
+        {from_: ts_4, to: ACTIVE_TS, "nk": 3, "c1": "baz", cname: "baz"},
+        {from_: ts_1, to: ts_2, "nk": 1, "c1": "foo", cname: "foo"},
+        {from_: ts_2, to: ACTIVE_TS, "nk": 1, "c1": "foo", cname: "foo_updated"},
+    ]
+
+
+@pytest.mark.parametrize(
+    "destination_config",
+    destinations_configs(default_sql_configs=True, subset=["duckdb"]),
+    ids=lambda x: x.name,
+)
+@pytest.mark.parametrize("simple", [True, False])
+def test_child_table(destination_config: DestinationTestConfiguration, simple: bool) -> None:
```

Comment:
i think we should test the child table functionality on all destinations, so either enable all dests here or add a child table to the basic test. the reason is, that we should run all sql code that we have against all destinations, just to make sure it works everywhere.

Reply:
Done in https://github.com/dlt-hub/dlt/pull/1168/commits/11748a63ffee50d2f01eb80a6138444a56743980